arangosh/Utils/ManagedDirectory.cpp

```diff
@@ -490,38 +488,31 @@ ssize_t ManagedDirectory::File::read(char* buffer, size_t length) {
     if (bytesRead < 0) {
       _status = _context->status();
     }
-  } else if (isGzip()) {
-    bytesRead = gzread(_gzFile, buffer, length);
-  } else {
-    bytesRead = ::rawRead(_fd, buffer, length, _status, _path, _flags);
+    return bytesRead;
   }
-#else
+#endif
   if (isGzip()) {
```

Comment:
If you are cleaning up read(), why not write() too.  Code does read better with your change.